include/mrdox/Plugin.hpp

```diff
@@ -12,16 +12,36 @@
 #define MRDOX_PLUGIN_HPP
 
 #include <mrdox/Platform.hpp>
-#include <mrdox/Generator.hpp>
+#include <mrdox/Support/Error.hpp>
+
+#include <string_view>
 
 namespace clang {
 namespace mrdox {
 
+class Generators;
+class Config;
+
+struct Plugin
+{
+    // MUST always come first
+    const std::string_view version;
+    Plugin(const std::string_view version) : version(version) {}
+
+    virtual
+    Error
+    registerGenerators(Generators & gen)
+    {
+        // noop is legal
+        return {};
+    }
+};
+
 extern "C"
 {
 
-// implement this function if you want to provide your own generator
-std::unique_ptr<Generator> makeMrDoxGenerator();
+// implement this function if you want to provide your own plugin
+Plugin & mrDoxMakePlugin(const Config & cfg);
```

Comment:
This function should not be necessary and it sure wouldn't be public?

Reply:
That's the plugin entry point, you described as `MrDoxPluginMain`.

Reply:
It is not a public API is what I am saying